privateca/certificate_authority_subordinate/main.tf

```diff
@@ -51,12 +50,14 @@ resource "google_privateca_certificate_authority" "root_ca" {
   key_spec {
     algorithm = "RSA_PKCS1_4096_SHA256"
   }
+  // Root CA should be valid indefinetly e.g. 99 years
+  lifetime = "${99 * 365 * 24 * 3600}s"
 }
 
 resource "google_privateca_certificate_authority" "default" {
   // This example assumes this pool already exists.
   // Pools cannot be deleted in normal test circumstances, so we depend on static pools
-  pool                     = "my-pool"
+  pool                     = "my-sub-pool"
```

Comment:
interesting; in the https://registry.terraform.io/providers/hashicorp/google/5.24.0/docs/resources/privateca_certificate_authority#example-usage---privateca-certificate-authority-subordinate example this uses the same pool name for both

Reply:
I believe technically the Google Cloud CAS product allows you to have different types of certificates in a single pool. But usually that does not make a whole lot of sense. Because in practice you also configure policy restrictions on a pool for issuing (in addition to the runtime constrains encoded in the certificate which are evaluated during usage by the tls library) and those will be fundamentally different for root certificates and subordinates.
Maybe for sample code that is used with automated testing it's useful to just stuff all certs into a single pool, but that is irritating for people aiming at building a working implementation.

Otherwise I might got a wrong understanding of how the CAS product and its pools should be used for the past two years. ;)

Reply:
Due to coincidence I just saw a new test case in the newly released TF provider which does exactly that: Setup a pool for the root and setup a pool for the subordinate:
https://github.com/hashicorp/terraform-provider-google-beta/pull/7197/files#diff-19706c53088b05bc2bc72f7bd03ee2eacbe6d12052408d3e905f150d2f3baafdR320

But it also configures EKU values on the root certificate, which is wrong there as well. I assume that is just wrong sample code that gets copied and pasted around now everywhere now.

Reply:
Agreed with @hoexter: in almost all cases, roots and subordinates should be in separate pools. The rationale behind CA Pools is that all CAs in the same pool should be completely interchangeable. So having 1 pool for the root CA(s) and another for the subordinate CA(s) is the way to go.

Reply:
This change is causing the tests to fail. This is due to the comment above this value, where this pool needs to already exist. 

I've created https://github.com/terraform-google-modules/terraform-docs-samples/pull/748 to support this change, which needs to be approved and merged before this PR will pass testing.

Reply:
Tests now pass with this change.